src/Symfony/Component/TypeInfo/TypeResolver/PhpDocAwareReflectionTypeResolver.php

```diff
@@ -33,13 +34,28 @@
  */
 final readonly class PhpDocAwareReflectionTypeResolver implements TypeResolverInterface
 {
+    private PhpDocParser $phpDocParser;
+    private Lexer $lexer;
+
     public function __construct(
         private TypeResolverInterface $reflectionTypeResolver,
         private TypeResolverInterface $stringTypeResolver,
         private TypeContextFactory $typeContextFactory,
-        private PhpDocParser $phpDocParser = new PhpDocParser(new TypeParser(), new ConstExprParser()),
-        private Lexer $lexer = new Lexer(),
+        ?PhpDocParser $phpDocParser = null,
+        ?Lexer $lexer = null,
     ) {
+        if (!$phpDocParser) {
```

Comment:
And what if I do pass a `$phpDocParser` to the constructor…?

Reply:
Sorry, kind of tired today... Thanks for taking care of that!